getTableSchema ignores database name

### Describe the bug
`getTableSchema(tableName, database)` should be able to obtain the table schema of a non-fully-qualified table name using the `database` second parameter, but it does not.

It instead returns this error:

```
DB::Exception: Table default.my_table does not exist. Maybe you meant my_database.my_table?. (UNKNOWN_TABLE) (version 25.2.1.1788 (official build))
```

### Steps to reproduce
`CREATE DATABASE my_database;`

Then create any table, `CREATE TABLE my_table (...)` using this template:

```java
var querySettings = new QuerySettings();
querySettings.setDatabase("my_database");
return clickHouseClient.query("CREATE TABLE my_table (...);", querySettings).get();
```

Then, try to register an entity (in my case, but just calling `getTableSchema` would work).

```
clickHouseClient.register(MyEntity.class, clickHouseClient.getTableSchema("my_table", "my_database"));
```

`getTableSchema` will produce an exception because it looks into the `default` database, not in `my_database`.

### Expected behaviour
`getTableSchema` should look in the database passed as an argument to resolve the table schema of a non-fully-qualified table name.

### Error log

```
Caused by: com.clickhouse.client.api.ServerException: Code: 60. DB::Exception: Table default.my_table does not exist. Maybe you meant my_database.my_table?. (UNKNOWN_TABLE) (version 25.2.1.1788 (official build)) 
	at app//com.clickhouse.client.api.internal.HttpAPIClientHelper.readError(HttpAPIClientHelper.java:359)
	at app//com.clickhouse.client.api.internal.HttpAPIClientHelper.executeRequest(HttpAPIClientHelper.java:407)
	at app//com.clickhouse.client.api.Client.lambda$query$10(Client.java:1622)
	at app//com.clickhouse.client.api.Client.runAsyncOperation(Client.java:1994)
	at app//com.clickhouse.client.api.Client.query(Client.java:1660)
	at app//com.clickhouse.client.api.Client.query(Client.java:1548)
	at app//com.clickhouse.client.api.Client.getTableSchemaImpl(Client.java:1873)
	at app//com.clickhouse.client.api.Client.getTableSchema(Client.java:1857)
```

### Note


I noticed this piece of code from the client:

```java
  private TableSchema getTableSchemaImpl(String describeQuery, String name, String originalQuery, String database) {
        int operationTimeout = getOperationTimeout();

        try (QueryResponse response = operationTimeout == 0 ? query(describeQuery).get() :
                query(describeQuery).get(getOperationTimeout(), TimeUnit.SECONDS)) {
            return TableSchemaParser.readTSKV(response.getInputStream(), name, originalQuery, database);
        } catch (TimeoutException e) {
            throw new ClientException("Operation has likely timed out after " + getOperationTimeout() + " seconds.", e);
        } catch (ExecutionException e) {
            throw new ClientException("Failed to get table schema", e.getCause());
        } catch (ServerException e) {
            throw e;
        } catch (Exception e) {
            throw new ClientException("Failed to get table schema", e);
        }
    }
```

The `query` method does not use any `QuerySettings` and does not set a database name from the parameters, thus

```java
public TableSchema getTableSchema(String table, String database) {
        final String sql = "DESCRIBE TABLE " + table + " FORMAT " + ClickHouseFormat.TSKV.name();
        return getTableSchemaImpl(sql, table, null, database);
    }
```

will try to describe a table against the `default` database. At least, this is my understanding at a cursory glance.


### Configuration
#### Environment
* Client version: 0.8.0
* Language version: java 17
* OS: mac os sequoia 15.2

#### ClickHouse server
* ClickHouse Server version: 25.2.1.1788
* ClickHouse Server non-default settings, if any:

--- a/client-v2/src/main/java/com/clickhouse/client/api/Client.java
+++ b/client-v2/src/main/java/com/clickhouse/client/api/Client.java
@@ -1882,8 +1882,9 @@ public class Client implements AutoCloseable {
     private TableSchema getTableSchemaImpl(String describeQuery, String name, String originalQuery, String database) {
         int operationTimeout = getOperationTimeout();
 
-        try (QueryResponse response = operationTimeout == 0 ? query(describeQuery).get() :
-                query(describeQuery).get(getOperationTimeout(), TimeUnit.SECONDS)) {
+        QuerySettings settings = new QuerySettings().setDatabase(database);
+        try (QueryResponse response = operationTimeout == 0 ? query(describeQuery, settings).get() :
+                query(describeQuery, settings).get(getOperationTimeout(), TimeUnit.SECONDS)) {
             return TableSchemaParser.readTSKV(response.getInputStream(), name, originalQuery, database);
         } catch (TimeoutException e) {
             throw new ClientException("Operation has likely timed out after " + getOperationTimeout() + " seconds.", e);

--- a/client-v2/src/test/java/com/clickhouse/client/insert/InsertTests.java
+++ b/client-v2/src/test/java/com/clickhouse/client/insert/InsertTests.java
@@ -105,8 +105,7 @@ public class InsertTests extends BaseIntegrationTest {
                 .useHttpCompression(useHttpCompression)
                 .setDefaultDatabase(ClickHouseServerForTest.getDatabase())
                 .serverSetting(ServerSettings.ASYNC_INSERT, "0")
-                .serverSetting(ServerSettings.WAIT_END_OF_QUERY, "1")
-                .useNewImplementation(System.getProperty("client.tests.useNewImplementation", "true").equals("true"));
+                .serverSetting(ServerSettings.WAIT_END_OF_QUERY, "1");
     }
 
     @AfterMethod(groups = { "integration" })
@@ -162,7 +161,7 @@ public class InsertTests extends BaseIntegrationTest {
         client.execute("DROP TABLE IF EXISTS " + tableName, commandSettings).get(EXECUTE_CMD_TIMEOUT, TimeUnit.SECONDS);
         client.execute(createSQL, commandSettings).get(EXECUTE_CMD_TIMEOUT, TimeUnit.SECONDS);
 
-        client.register(PojoWithJSON.class, client.getTableSchema(tableName, "default"));
+        client.register(PojoWithJSON.class, client.getTableSchema(tableName));
         PojoWithJSON pojo = new PojoWithJSON();
         pojo.setEventPayload(originalJsonStr);
         List<Object> data = Arrays.asList(pojo);
@@ -189,7 +188,7 @@ public class InsertTests extends BaseIntegrationTest {
 
         initTable(tableName, createSQL);
 
-        client.register(SamplePOJO.class, client.getTableSchema(tableName, "default"));
+        client.register(SamplePOJO.class, client.getTableSchema(tableName));
 
         System.out.println("Inserting POJO: " + pojo);
         try (InsertResponse response = client.insert(tableName, Collections.singletonList(pojo), settings).get(EXECUTE_CMD_TIMEOUT, TimeUnit.SECONDS)) {
@@ -231,7 +230,7 @@ public class InsertTests extends BaseIntegrationTest {
 
         initTable(tableName, createSQL);
 
-        client.register(SamplePOJO.class, client.getTableSchema(tableName, "default"));
+        client.register(SamplePOJO.class, client.getTableSchema(tableName));
 
         try (InsertResponse response = client.insert(tableName, Collections.singletonList(pojo), settings).get(30, TimeUnit.SECONDS)) {
             fail("Should have thrown an exception");

--- a/client-v2/src/test/java/com/clickhouse/client/metadata/MetadataTests.java
+++ b/client-v2/src/test/java/com/clickhouse/client/metadata/MetadataTests.java
@@ -14,7 +14,9 @@ import com.clickhouse.client.api.enums.Protocol;
 import com.clickhouse.client.api.internal.ServerSettings;
 import com.clickhouse.client.api.metadata.DefaultColumnToMethodMatchingStrategy;
 import com.clickhouse.client.api.metadata.TableSchema;
+import com.clickhouse.client.api.query.QuerySettings;
 import com.clickhouse.data.ClickHouseColumn;
+import com.clickhouse.data.ClickHouseDataType;
 import com.clickhouse.data.ClickHouseRecord;
 import org.testng.Assert;
 import org.testng.annotations.BeforeMethod;
@@ -47,6 +49,24 @@ public class MetadataTests extends BaseIntegrationTest {
         List<ClickHouseColumn> columns = schema.getColumns();
         Assert.assertEquals(columns.get(0).getColumnName(), "param1");
         Assert.assertEquals(columns.get(0).getDataType().name(), "UInt32");
+    }
+
+    @Test(groups = { "integration" })
+
+    public void testGetTableSchemaDifferentDb() throws Exception {
+        String table = "test_get_table_schema_different_db";
+        String db = ClickHouseServerForTest.getDatabase() + "_schema_test" ;
+        try {
+            QuerySettings settings = new QuerySettings().setDatabase(db);
+            client.execute("DROP DATABASE IF EXISTS " + db).get().close();
+            client.execute("CREATE DATABASE " + db).get().close();
+            client.query("DROP TABLE IF EXISTS " + table, settings).get().close();
+            client.query("CREATE TABLE " + table + " (rowId Int32) Engine=MergeTree ORDER BY ()", settings).get().close();
+            TableSchema tableSchema = client.getTableSchema(table, db);
+            Assert.assertEquals(tableSchema.getColumnByName("rowId").getDataType(), ClickHouseDataType.Int32);
+        } finally {
+            client.execute("DROP DATABASE IF EXISTS " + db).get().close();
+        }
     }
 
     private void prepareDataSet(String tableName) {

--- a/client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java
+++ b/client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java
@@ -1883,7 +1883,7 @@ public class QueryTests extends BaseIntegrationTest {
             throw e;
         }
 
-        client.register(AggregateFuncDTO.class, client.getTableSchema(tableName, "default"));
+        client.register(AggregateFuncDTO.class, client.getTableSchema(tableName));
 
         try (InsertResponse response = client.insert(tableName, Collections.singletonList(pojo)).get(30, TimeUnit.SECONDS)) {
             Assert.assertEquals(response.getWrittenRows(), 1);

